resolver/src/lookup_state.rs

```diff
@@ -239,61 +240,54 @@ impl<C: DnsHandle<Error = ResolveError> + 'static> QueryFuture<C> {
 
     fn handle_noerror(&mut self, mut message: Message) -> Poll<Records, ResolveError> {
         // seek out CNAMES
```

Comment:
If the query was a CNAME or ALL query, then we shouldn't do this CNAME processing, right? A CNAME (or ALL) query should return the CNAME itself, right?

Reply:
Oh. that's a good point. I think you are correct.